### ISSUE TITLE
How did you adapt RENet (ICRA’23, MOD task) for fair comparison in EOLO?

### ISSUE THREAD

**Issue by lhqlhqlhq123** [AndyCao1125/EOLO#4]
Hi, thanks for releasing EOLO and the synthetic datasets!

I’m trying to reproduce the comparisons with RENet reported in your paper, but I ran into a conceptual gap:

RENet (Zhou et al., ICRA’23) is designed for Moving Object Detection (MOD) on DSEC-MOD, with a different head/loss and metrics (F.mAP / V.mAP).

EOLO performs standard object detection (mAP50/75) under all-day exposure conditions.

Which parts of the official RENet repo did you change?

**Comment by SageCao1125** [comment]
@lhqlhqlhq123  👍 

Thank you for your question!

Indeed, we have made some modifications to the original code, and we would like to clearly explain the differences in the revised version of the code that we used for the experiments.

## Differences between MOD_Net and the Modified Version (MOD_Net_Yolo)

1. **Input Features:**
   - The original MOD_Net directly processes raw image sequences with a heavy backbone network. In contrast, the modified version, MOE_Net_Yolo, takes both RGB and event-based features as input.  
   - **Reason:** The primary goal of this modification is to evaluate the effectiveness of the fusion module. Therefore, we aimed to keep the input features consistent, as the main comparison lies in the fusion module’s performance.

2. **Input Structure:**
   - Based on the above change, we now input N pairs of RGB-event embeddings directly. As a result, the corresponding chunks are reduced to only 3.

3. **MoE Branch Module:**
   - We have omitted the last MoE branch module used in the original MOD_Net. This decision is aligned with your claim, as the head module in MOD_Net is primarily designed for motion detection. On the other hand, our static detection uses the output features as C3, C4, C5 for the subsequent YOLO detection head. This approach is consistent with the YOLO architecture.

## Summary

To summarize, you can understand our approach as follows: while replicating the core components of RENet, we retained the key module proposed by the authors, the Bi-Directional Calibration (BDC) layer, for feature fusion. The output, C3, C4, C5, is predicted as:

```
C3, C4, C5 = MoD_Net(BDC_layer=3, input=[image_emb, event_emb]) 
```

In our EOLO model, this is achieved through:

```
C3, C4, C5 = EOLO_fusion(input=[image_emb, event_emb])
```
Both outputs are then fed into the same detection head for prediction.


Thank you again for bringing up this important point! 

I sincerely apologize for this confusion. It serves as a reminder for me to properly document these modifications in the repository, ensuring transparency and fairness.

**Comment by SageCao1125** [comment]
## Revised MoE_Net:
```
class MOD_Net_yolo(nn.Module):
    def __init__(self):
        super(MOD_Net_yolo, self).__init__()
        

        self.middle_fusion_1_ful = nn.Sequential(nn.Conv2d(128, 128, kernel_size=3, stride=1, padding=1, bias=True), nn.BatchNorm2d(128, momentum=BN_MOMENTUM), nn.ReLU(inplace=True))
        self.middle_fusion_2_ful = nn.Sequential(nn.Conv2d(128+256, 256, kernel_size=3, stride=1, padding=1, bias=True), nn.BatchNorm2d(256, momentum=BN_MOMENTUM), nn.ReLU(inplace=True))
        self.middle_fusion_3_ful = nn.Sequential(nn.Conv2d(256+512, 512, kernel_size=3, stride=1, padding=1, bias=True), nn.BatchNorm2d(512, momentum=BN_MOMENTUM), nn.ReLU(inplace=True))
        
        ## fuison
        self.fus = nn.ModuleList([
            REFusion(128, 128, 0),
            REFusion(256, 256, 0),
            REFusion(512, 512, 0),
        ])
        self.maxpool = nn.MaxPool2d(kernel_size=3, stride=2, padding=1)

    
    def forward(self, RGB_f, event_f):
        img_c3, img_c4, img_c5 = RGB_f      ## [N,128,40,40], [N,256,20,20], [N,512,10,10]
        e_c3, e_c4, e_c5 = event_f

        ## fuison
        
        chunk_1 = self.fus[0](img_c3, e_c3)
        chunk_1 = self.middle_fusion_1_ful(chunk_1)
        
        chunk_2 = self.fus[1](img_c4, e_c4)
        chunk_2 = torch.cat((chunk_2, self.maxpool(chunk_1)), dim=1)
        chunk_2 = self.middle_fusion_2_ful(chunk_2)
        
        chunk_3 = self.fus[2](img_c5, e_c5)
        chunk_3 = torch.cat((chunk_3, self.maxpool(chunk_2)), dim=1)
        chunk_3 = self.middle_fusion_3_ful(chunk_3)

        return [chunk_1, chunk_2, chunk_3]
```
## Main:
```
class EYOLOTiny(nn.Module):
    def __init__():
    ...
    if self.fusion_method == 'RENet':
       self.fusion_mode_and_level = MOD_Net_yolo()

    ... Others are the same as EOLO's.
    
    def forward():
    ...
       if self.fusion_method == 'RENet':
          c3, c4, c5 = self.fusion_mode_and_level([img_c3, img_c4, img_c5], [event_c3, event_c4, event_c5])

    ... Others are the same as EOLO's.
```

**Comment by SageCao1125** [comment]
@lhqlhqlhq123 

Hope the codes can help you understand.

Thanks!
